### ISSUE TITLE
[Question] Space Station dimension time

### ISSUE THREAD

**Issue by 016Nojr** [zmaster587/AdvancedRocketry#1273]
How is the time in the space station calculated? I thought it was always day, but some solar panels in there detects it as night. Is it always night?

**Comment by C1ffisme** [comment]
Apparently the Solar Panel detects if it is daytime using `world.isDaytime()`.

Here's the line: https://github.com/zmaster587/AdvancedRocketry/blob/1.12/src/main/java/zmaster587/advancedRocketry/tile/TileSolarPanel.java#L22

I'm not really sure where `world.isDaytime()` gets the day/night cycle from, but I assume it's not what's regulating the celestial body positions or the sunlight on a spacestation.

**Comment by scotty9090** [comment]
I'm running into this with Mekanism Advanced Solar's as well.  They seem to be stuck at night time now.  Changing the MC time doesn't seem to be affecting anything.

**Comment by agent55511** [comment]
For reference Mekansim Solar Generators appear to check for the sun using this line of code:
https://github.com/aidancbrady/Mekanism/blob/5d8b5c8f915b1f435e156ef3131a1b72d44e2e89/src/main/java/mekanism/generators/common/tile/TileEntitySolarGenerator.java#L66

They also appear to use the `world.isDaytime()` in its sun checking code.

**Comment by zmaster587** [comment]
It seems I am a bit late to the party, but isDaytime is calculated by getSkylightSubtracted, which in turn is calculated from getSunBrightnessFactor which in turn is calculated from the angle of the sun, which is determined by the space station's rotation.

TL;DR: time is calculated from the rotation of the station which is visually indicated by the position of the sun.
